test/test_type_promotion.py

```diff
@@ -79,15 +79,26 @@ def test_int_promotion(self, device):
         self.assertEqual(c.dtype, torch.int64)
 
     @float_double_default_dtype
-    def test_float_promotion(self, device):
-        a = torch.ones([4, 4, 4], dtype=torch.float, device=device)
-        b = torch.ones([4, 4, 4], dtype=torch.double, device=device)
-        c = a + b
-        self.assertEqual(c, b + b)
-        self.assertEqual(c.dtype, torch.double)
-        c = b + a
-        self.assertEqual(c, b + b)
-        self.assertEqual(c.dtype, torch.double)
+    def test_complex_float_promotion(self, device):
+        def test_promotion(dtype_float, dtype_double):
+            a = torch.ones([4, 4, 4], dtype=dtype_float, device=device)
+            b = torch.ones([4, 4, 4], dtype=dtype_double, device=device)
+            c = a + b
+            self.assertEqual(c, b + b)
+            self.assertEqual(c.dtype, dtype_double)
+            c = b + a
+            self.assertEqual(c, b + b)
+            self.assertEqual(c.dtype, dtype_double)
+        test_promotion(torch.float, torch.double)
+        test_promotion(torch.complex64, torch.complex128)
```

Comment:
@mruberry Can you suggest how to skip complex dtypes entirely for XLA? ;) Like the half types?

Reply:
Try adding the unsupported types here: https://github.com/pytorch/xla/blob/c77641f460133b0f1c75ac8c3533b987278b5c25/torch_patches/X10-device_test.diff#L46

Reply:
Hmmm that only works when dtype appears in the decorator and it's skipped in instantiation of tests. But here we explicitly call with complex dtypes so it doesn't work. (please correct me if I'm wrong). 
@anjali411 would you mind refactor these tests to keep float and complex in separate tests? They can call into common helper functions. Since type promotion is pretty major behavior, I do want to keep XLA running the part they're able to run. Thanks a lot!